packages/repack/src/modules/ScriptManager/ScriptManager.ts

```diff
@@ -329,6 +329,43 @@ export class ScriptManager extends EventEmitter {
     }
   }
 
+  /**
+   * Loads a script with retry logic.
+   *
+   * This function attempts to load a script using the nativeScriptManager.
+   * If the initial attempt fails, it retries the specified number of times
+   * with an optional delay between retries.
+   *
+   * @param {string} scriptId - The ID of the script to load.
+   * @param {NormalizedScriptLocator} locator - An NormalizedScriptLocator containing retry configuration.
+   * @param {number} [locator.retry=0] - The number of retry attempts.
+   * @param {number} [locator.retryDelay=0] - The delay in milliseconds between retries.
+   * @throws {Error} Throws an error if all retry attempts fail.
+   */
+  async loadScriptWithRetry(
+    scriptId: string,
+    locator: NormalizedScriptLocator
+  ) {
+    const { retry = 0, retryDelay = 0 } = locator;
+    let attempts = retry + 1; // Include the initial attempt
+
+    while (attempts > 0) {
+      try {
+        await this.nativeScriptManager.loadScript(scriptId, locator);
+        return; // Successfully loaded the script, exit the loop
+      } catch (error) {
+        attempts--;
+        if (attempts > 0) {
+          if (retryDelay > 0) {
+            await waitMs(retryDelay);
```

Comment:
no need to make this a separate function here, lets just inline this

Reply:
Done.